src/chi/javascript/components/mobile-navigation.js

```diff
@@ -0,0 +1,458 @@
+      drawerMenuItemList.style.removeProperty('display');
+      drawerMenuItemList.style.removeProperty('visibility');
+      drawerMenuItemList.style.removeProperty('position');
+    }
+
+    if (!Util.checkHasClass(drawerMenuItem, DRAWER_ITEM_LIST_EXPANDED)) {
+      this._menuItemAnimation = Util.threeStepsAnimation(
+        function() {
+          calculateHeight();
+          Util.checkAddClass(drawerMenuItemList, chi.classes.TRANSITIONING);
+          drawerMenuItemList.style.height = '0px';
+        }, function() {
+          drawerMenuItemList.style.height = drawerMenuItemListHeight;
+          Util.checkAddClass(drawerMenuItem, DRAWER_ITEM_LIST_EXPANDED);
+        }, function() {
+          Util.checkRemoveClass(drawerMenuItemList, chi.classes.TRANSITIONING);
+        }, 75
+      );
+    } else {
+      this._menuItemAnimation = Util.threeStepsAnimation(
+        function() {
+          calculateHeight();
+          Util.checkAddClass(drawerMenuItemList, chi.classes.TRANSITIONING);
+          drawerMenuItemList.style.height = drawerMenuItemListHeight;
+        }, function() {
+          drawerMenuItemList.style.height = '0px';
+          Util.checkRemoveClass(drawerMenuItem, DRAWER_ITEM_LIST_EXPANDED);
+        }, function() {
+          Util.checkRemoveClass(drawerMenuItemList, chi.classes.TRANSITIONING);
+          drawerMenuItemList.style.removeProperty('height');
+        }, 75
+      );
+    }
+  }
+
+  resetActiveDrawerMenuItem() {
+    const drawerExpandedMenuItem = this._getExpandedSecondLevelItem();
+    const currentlyActiveDrawerItemSubtab = this._getActiveSecondLevelItemSubtab();
+
+    this.drawerMenuItemRemoveUnselected();
+    if (drawerExpandedMenuItem) {
+      const drawerExpandedMenuItemList = drawerExpandedMenuItem.querySelector(`.${DRAWER_ITEM_LIST_CLASS}`);
+
+      if (drawerExpandedMenuItemList &&
+        !drawerExpandedMenuItemList.querySelector(`ul li a.${chi.classes.ACTIVE}`)) {
+        drawerExpandedMenuItem.querySelector(`.${DRAWER_ITEM_LIST_CLASS}`).style.removeProperty('height');
+        Util.checkRemoveClass(drawerExpandedMenuItem, chi.classes.EXPANDED);
+      }
+    }
+
+    if (currentlyActiveDrawerItemSubtab) {
+      let drawerMenuItemToActivate;
+
+      for (let cur = currentlyActiveDrawerItemSubtab;
+           cur && !Util.checkHasClass(cur, DRAWER_NAV_LIST);
+           cur = cur.parentNode) {
+        drawerMenuItemToActivate = cur;
+      }
+
+      Util.checkAddClass(drawerMenuItemToActivate, chi.classes.ACTIVE);
+      Util.checkAddClass(drawerMenuItemToActivate, chi.classes.EXPANDED);
+    }
+  }
+
+  _locateMobileNav() {
+    const mobileNavElem = Util.getTarget(this._elem);
+    if (!mobileNavElem) {
+      throw new Error('Could not find Mobile Navigation content for trigger. ');
+    }
+    return mobileNavElem;
+  }
+
+  dispose() {
+    this._config = null;
+    this._elem = null;
+    this._firstLevelDrawer = null;
```

Comment:
Please call this._firstLevelDrawer.dispose() first.

Reply:
Ensure that you dispose all the components that you instantiate when they are not needed anymore. If you don't do this, it will became a memory leak.